distributed/scheduler.py

```diff
@@ -50,6 +50,7 @@
     valmap,
 )
 from tornado.ioloop import IOLoop
+from typing_extensions import Self
```

Comment:
```suggestion
if TYPE_CHECKING:
    from typing_extensions import Self
```
not a runtime dependency